File: lib/models/log_entry.dart
enum LogLevel {
  info('INFO', 'Information'),
  warning('WARNING', 'Warning'),
  error('ERROR', 'Error'),
  success('SUCCESS', 'Success');

  const LogLevel(this.displayName, this.description);

  final String displayName;
  final String description;
}

class LogEntry {
  final DateTime timestamp;
  final LogLevel level;
  final String message;
  final String? source;

  const LogEntry({
    required this.timestamp,
    required this.level,
    required this.message,
    this.source,
  });

  factory LogEntry.info(String message, {String? source}) {
    return LogEntry(
      timestamp: DateTime.now(),
      level: LogLevel.info,
      message: message,
      source: source,
    );
  }

  factory LogEntry.warning(String message, {String? source}) {
    return LogEntry(
      timestamp: DateTime.now(),
      level: LogLevel.warning,
      message: message,
      source: source,
    );
  }

  factory LogEntry.error(String message, {String? source}) {
    return LogEntry(
      timestamp: DateTime.now(),
      level: LogLevel.error,
      message: message,
      source: source,
    );
  }

  factory LogEntry.success(String message, {String? source}) {
    return LogEntry(
      timestamp: DateTime.now(),
      level: LogLevel.success,
      message: message,
      source: source,
    );
  }

  String get formattedTimestamp {
    return '${timestamp.hour.toString().padLeft(2, '0')}:'
        '${timestamp.minute.toString().padLeft(2, '0')}:'
        '${timestamp.second.toString().padLeft(2, '0')}.'
        '${timestamp.millisecond.toString().padLeft(3, '0')}';
  }

  String get formattedMessage {
    final prefix = source != null ? '[$source] ' : '';
    return '[$formattedTimestamp] [${level.displayName}] $prefix$message';
  }

  Map<String, dynamic> toJson() {
    return {
      'timestamp': timestamp.toIso8601String(),
      'level': level.name,
      'message': message,
      'source': source,
    };
  }

  factory LogEntry.fromJson(Map<String, dynamic> json) {
    return LogEntry(
      timestamp: DateTime.parse(json['timestamp']),
      level: LogLevel.values.firstWhere((e) => e.name == json['level']),
      message: json['message'],
      source: json['source'],
    );
  }

  @override
  String toString() {
    return formattedMessage;
  }
}

File: android/app/src/main/kotlin/com/binah/cryptingtool/models/LogEntry.kt
package com.binah.cryptingtool.models

import android.os.Parcelable
import kotlinx.parcelize.Parcelize
import java.util.Date

enum class LogLevel(val displayName: String, val description: String) {
    INFO("INFO", "Information"),
    WARNING("WARNING", "Warning"),
    ERROR("ERROR", "Error"),
    SUCCESS("SUCCESS", "Success")
}

@Parcelize
data class LogEntry(
    val timestamp: Date = Date(),
    val level: LogLevel,
    val message: String,
    val source: String? = null
) : Parcelable {
    companion object {
        fun info(message: String, source: String? = null) = LogEntry(
            level = LogLevel.INFO,
            message = message,
            source = source
        )
        
        fun warning(message: String, source: String? = null) = LogEntry(
            level = LogLevel.WARNING,
            message = message,
            source = source
        )
        
        fun error(message: String, source: String? = null) = LogEntry(
            level = LogLevel.ERROR,
            message = message,
            source = source
        )
        
        fun success(message: String, source: String? = null) = LogEntry(
            level = LogLevel.SUCCESS,
            message = message,
            source = source
        )
    }
    
    fun getFormattedTime(): String {
        val now = Date()
        val difference = now.time - timestamp.time
        val seconds = difference / 1000
        val minutes = seconds / 60
        val hours = minutes / 60
        
        return when {
            hours > 0 -> "${hours}h ago"
            minutes > 0 -> "${minutes}m ago"
            seconds > 10 -> "${seconds}s ago"
            else -> "Just now"
        }
    }
}